raiden_contracts/data/source/services/UserDeposit.sol

```diff
@@ -167,23 +167,37 @@ contract UserDeposit is Utils {
     /// Removes the withdraw plan even if not the full amount has been
     /// withdrawn.
     /// @param amount Amount of tokens to be withdrawn
-    function withdraw(uint256 amount)
+    /// @param beneficiary Address to send withdrawn tokens to
+    function withdrawToBeneficiary(uint256 amount, address beneficiary)
         external
     {
-        WithdrawPlan storage withdraw_plan = withdraw_plans[msg.sender];
+        require(beneficiary != address(0x0), "beneficiary is zero");
+        WithdrawPlan storage withdraw_plan = withdraw_plans[tx.origin];
```

Comment:
Those changes (`msg.sender` to `tx.origin`) are not super nice, but they seem ok here.